app/src/main/java/com/asma/tasky/feature_management/domain/notification_service/ReminderNotificationService.kt

```diff
@@ -0,0 +1,50 @@
+package com.asma.tasky.feature_management.domain.notification_service
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.TaskStackBuilder
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import com.asma.tasky.R
+import com.asma.tasky.core.presentation.MainActivity
+import com.asma.tasky.feature_management.domain.AgendaItem
+
+class ReminderNotificationService (private val context: Context) {
```

Comment:
This is not a domain related class, the domain layer should be isolated from all foreign dependencies which also counts for Android specific ones (like context and `NotificationManager` here).
This class won't be locally unit testable since it contains such Android related dependencies. The fix is to create an interface with these functions in domain and then have this class as the implementation in data.